### PR TITLE
asserts: add Model.AllSnaps method

### PR DESCRIPTION
**Pull request by andrewphelpsj**
This method is a small helper that will be used during the creation of recovery systems. It also simplifies another case during remodeling a little bit.

### CODE REVIEW

File: asserts/model_test.go [hunk 1]
```diff
@@ -1407,3 +1407,19 @@ func (mods *modelSuite) TestModelValidationSetSequenceKey(c *C) {
...
+	encoded = strings.Replace(encoded, "OTHER", "", 1)
+
+	a, err := asserts.Decode([]byte(encoded))
+	c.Assert(err, IsNil)
+	c.Check(a.Type(), Equals, asserts.ModelType)
+
+	model := a.(*asserts.Model)
+
+	allSnaps := append([]*asserts.ModelSnap(nil), model.EssentialSnaps()...)
+	allSnaps = append(allSnaps, model.SnapsWithoutEssential()...)
```
Comment (pedronis) [inline asserts/model_test.go:1422]: are both of these non-empty here?

File: asserts/model.go [hunk 2]
```diff
@@ -613,6 +613,14 @@ func (mod *Model) SnapsWithoutEssential() []*ModelSnap {
 	return mod.allSnaps[mod.numEssentialSnaps:]
 }
 
+// AllSnaps returns all the snaps listed by the model. Essential snaps are at
```
Comment (pedronis) [inline asserts/model.go:616]: "... by the model across all modes." perhaps?

**Review by pedronis — APPROVED** [review]
thanks, small remarks

File: asserts/model_test.go (hunk 1 above)
Comment (andrewphelpsj) [reply asserts/model_test.go:1422]: Yes, they are. But good point, I'll add some assertions to make sure that the test is actually doing something useful.

**Review by Meulengracht — APPROVED** [review]
Thank you, lgtm

**Comment by ernestl** [comment]
Test failures does not relate to changes. Force merging.

**Pull request merged**